Node.js/sampleresponse.js

```diff
@@ -40,14 +40,16 @@ var mqBoilerPlate = new MQBoilerPlate();
 
 async function msgCB(md, buf) {
```

Comment:
msgCB is an async process : which means that it is returning a promise. That means that the callback passed into `mqBoilerPlate.getMessages` could be receiving either a normal function or a function that returns a promise. 

That means that in boilerplate:

```
  getCallback(err, hObj, gmo, md, buf) {
    // If there is an error, prepare to exit by setting the ok flag to false.
    if (err) {
      if (err.mqrc == MQC.MQRC_NO_MSG_AVAILABLE) {
        debug_info("No more messages available.");
      } else {
        MQBoilerPlate.reportError(err);
      }
      canExit = true;
      // We don't need any more messages delivered, so cause the
      // callback to be deleted after this one has completed.
      //mq.GetDone(hObj);
    } else {
      if (activeCB) {
        canExit = !activeCB(md, buf);
      }
    }
  }
  ```

The lines
```
      if (activeCB) {
        canExit = !activeCB(md, buf);
      }
      ```
      
      Needs to check if the return is a promise or a value. ie. 
      
      ```
      if (activeCB) {
        if (activeCB instanceof Promise) {
           activeCB(md, buf)
           .then((mustContinue) => {canExit = ! mustContinue;})
        } else {
           canExit = !activeCB(md, buf);
        }
      }
      ```

This might fix the need to use process.exit()